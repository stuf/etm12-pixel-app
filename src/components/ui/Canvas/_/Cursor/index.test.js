import * as React from 'karet';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Index from './index';

it('calculates the correct position', () => {
  const pos = [2, 4];
  const scale = 16;

  const wrap = shallow(<Index {...{ pos, scale }} />);

  expect(wrap.prop('style')).toEqual({
    width: scale,
    height: scale,
    transform: `translateX(${pos[0] * scale}px) translateY(${pos[1] *
      scale}px)`,
  });
});

it('matches snapshot', () => {
  const pos = [2, 4];
  const scale = 16;

  const wrap = shallow(<Index {...{ pos, scale }} />);

  expect(toJson(wrap)).toMatchSnapshot();
});
