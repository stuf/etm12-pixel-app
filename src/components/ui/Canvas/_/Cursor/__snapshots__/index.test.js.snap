// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="root"
  style={
    Object {
      "height": 16,
      "transform": "translateX(32px) translateY(64px)",
      "width": 16,
    }
  }
/>
`;
